aten/src/ATen/native/cpu/ActivationKernel.cpp

```diff
@@ -0,0 +1,484 @@
+#include "ATen/native/cpu/ActivationKernel.h"
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+#include <numeric>
+
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include "ATen/cpu/vec256/vec256.h"
+#include "ATen/optional.h"
+
+// NOTE: In general we avoid calls into cmath for code compiled with AVX/AVX2
+// This is because of SSE-AVX transitions and a bug in Glibc2.23
+// See https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1663280
+
+namespace at { namespace native {
+namespace {
+
+static tbb::affinity_partitioner ap;
+
+template <int64_t size>
+inline int64_t _leftover(int64_t x, int64_t y) {
+  if (x + size > y)
+    return y - x;
+  return size;
+}
+
+template <typename scalar_t>
+inline scalar_t _vec_sum(int64_t dim_size, scalar_t* input_data) {
+  using Vec = vec256::Vec256<scalar_t>;
+  for (int ii = 0; ii < dim_size; ii++) {
+  }
+  int64_t d = 0;
+  Vec sum_vec(0);
+  scalar_t sum = 0;
+  Vec value(0);
+  for (; d < dim_size; d += Vec::size) {
+    if (d + Vec::size > dim_size) {
+      // TODO: For some reason, when using partial loads, value isn't set to 0
+      // on each iteration
+      for (int64_t i = d; i < dim_size; i++) {
+        sum += input_data[i];
+      }
+    } else {
+      value.load(input_data + d);
+      sum_vec = sum_vec + value;
+    }
+  }
```

Comment:
Have you look at the assembly generated for this part? I'd like to be sure that we don't do the `d + Vec::size > dim_size` comparison at every iteration. It might be better to just have this loop until `dim_size - (dim_size % Vec::size)` and then an epilogue that will process last `dim_size % Vec::size` elems.

Reply:
Right. This is something that needs to be debugged. It should work.